proof-producer/bin/proof-producer/include/nil/proof-generator/prover.hpp

```diff
@@ -26,10 +26,15 @@
 #include <random>
 #include <sstream>
 #include <optional>
-#include <chrono>
 
 #include <boost/log/trivial.hpp>
 
+#ifndef TIME_LOG_ENABLED
```

Comment:
I think you can remove this ifndef, the file scoped_profiler.hpp has include guards inside anyway.

Reply:
It may be useful in case allow profiling only for some module.
Some times turn on in assigner library for example

Reply:
I don't get how this helps, you don't undef it at the end anyway...